### PR TITLE
CB-235: Tags on release-group page

### PR DESCRIPTION
**Pull request by ferbncode**
For ticket CB-235
Presently, tags are not shown on the release group page. Tags can be shown on the release group page under the release group information.

All tags are shown similar to (with their links pointing to musicbrainz.org/tag/tagname):

and:

Please suggest improvements :)

### CODE REVIEW

**Comment by gentlecat** [comment]
@brainzbot, ok to test.

**Comment by gentlecat** [comment]
Next time if PR is related to a specific ticket, mention it in a title like I did.

File: critiquebrainz/frontend/external/musicbrainz.py [hunk 1]
```diff
@@ -33,6 +33,7 @@ def search_release_groups(query='', artist='', release_group='', limit=None, off
     return api_resp.get('release-group-count'), api_resp.get('release-group-list')
 
 
+
```
Comment (gentlecat) [inline critiquebrainz/frontend/external/musicbrainz.py]: This seems unnecessary.

File: critiquebrainz/frontend/templates/macros.html [hunk 2]
```diff
@@ -93,6 +93,19 @@
   </div>
 {%- endmacro %}
 
+{% macro show_tags(tags) %}
+    <div class="tags">
+        {% if tags %}
+          {% for i in range(tags|length - 1) %}
+          <span><a style="font-size:12px" href="{{"https://musicbrainz.org/tag/%s"|format(tags[i]['name'])}}"> {{tags[i]['name']}}, </a></span>
```
Comment (gentlecat) [inline critiquebrainz/frontend/templates/macros.html]: This should be indented. Also, we use 2 spaces for indentation in templates.

File: critiquebrainz/frontend/templates/macros.html [hunk 3]
```diff
@@ -93,6 +93,19 @@
   </div>
 {%- endmacro %}
 
+{% macro show_tags(tags) %}
+    <div class="tags">
+        {% if tags %}
+          {% for i in range(tags|length - 1) %}
```
Comment (gentlecat) [inline critiquebrainz/frontend/templates/macros.html]:
```python
for tag in tags
```

File: critiquebrainz/frontend/templates/release_group/entity.html [hunk 4]
```diff
@@ -156,6 +156,12 @@ <h4 style="margin-bottom:0;">{{ _('Reviews') }}</h4>
         {% endfor %}
       </ul>
     {% endif %}
+    
+    <div class="release-group-tags">
+        <b>{{ _('Tags') }}</b>
+        {{ show_tags(tags) }}
+        <p></p>
```
Comment (gentlecat) [inline critiquebrainz/frontend/templates/release_group/entity.html]: What's this for?

File: critiquebrainz/frontend/templates/release_group/entity.html [hunk 5]
```diff
@@ -156,6 +156,12 @@ <h4 style="margin-bottom:0;">{{ _('Reviews') }}</h4>
         {% endfor %}
       </ul>
     {% endif %}
+    
+    <div class="release-group-tags">
```
Comment (gentlecat) [inline critiquebrainz/frontend/templates/release_group/entity.html]: What's the purpose of this tag and its `release-group-tags` class?

File: critiquebrainz/frontend/templates/macros.html [hunk 6]
```diff
@@ -93,6 +93,19 @@
...
+          <span><a style="font-size:12px" href="{{"https://musicbrainz.org/tag/%s"|format(tags[i]['name'])}}"> {{tags[i]['name']}}, </a></span>
+          {% endfor %}
+          <span><a style="font-size:12px" href="{{"https://musicbrainz.org/tag/%s"|format(tags[tags|length-1]['name'])}}"> {{tags[tags|length -1]['name']}} </a></span>
+        {% else %}
+           <p style="font-size:12px">(none)</p>
```
Comment (gentlecat) [inline critiquebrainz/frontend/templates/macros.html]: `(none)` string should be translatable.

File: critiquebrainz/frontend/templates/macros.html [hunk 7]
```diff
@@ -93,6 +93,19 @@
   </div>
 {%- endmacro %}
 
+{% macro show_tags(tags) %}
+    <div class="tags">
+        {% if tags %}
+          {% for i in range(tags|length - 1) %}
+          <span><a style="font-size:12px" href="{{"https://musicbrainz.org/tag/%s"|format(tags[i]['name'])}}"> {{tags[i]['name']}}, </a></span>
+          {% endfor %}
+          <span><a style="font-size:12px" href="{{"https://musicbrainz.org/tag/%s"|format(tags[tags|length-1]['name'])}}"> {{tags[tags|length -1]['name']}} </a></span>
```
Comment (gentlecat) [inline critiquebrainz/frontend/templates/macros.html]: Custom styling (like `font-size:12px`) should go into Less files instead of templates.

File: critiquebrainz/frontend/templates/macros.html [hunk 8]
```diff
@@ -93,6 +93,19 @@
   </div>
 {%- endmacro %}
 
+{% macro show_tags(tags) %}
+    <div class="tags">
```
Comment (gentlecat) [inline critiquebrainz/frontend/templates/macros.html]: Is the `tags` class referenced anywhere?

File: critiquebrainz/frontend/templates/macros.html [hunk 9]
```diff
@@ -93,8 +93,19 @@
...
 {%- endmacro %}
 
+{% macro show_tags(tags) %}
+  {% if tags %}
+    {% for tag in tags[: -1] %}
+      <span><a href="{{"https://musicbrainz.org/tag/%s"|format(tag['name'])}}"> {{tag['name']}}, </a></span>
+    {% endfor %}
+    <span><a href="{{"https://musicbrainz.org/tag/%s"|format(tags[-1]['name'])}}"> {{tags[-1]['name']}} </a></span>
+  {% else %}
+  <p>{{ _('(none)') }}</p>
```
Comment (gentlecat) [inline critiquebrainz/frontend/templates/macros.html]: Indentation is off here.

File: critiquebrainz/frontend/templates/macros.html [hunk 10]
```diff
@@ -93,8 +93,19 @@
   </div>
 {%- endmacro %}
 
+{% macro show_tags(tags) %}
+  {% if tags %}
+    {% for tag in tags[: -1] %}
+      <span><a href="{{"https://musicbrainz.org/tag/%s"|format(tag['name'])}}"> {{tag['name']}}, </a></span>
+    {% endfor %}
+    <span><a href="{{"https://musicbrainz.org/tag/%s"|format(tags[-1]['name'])}}"> {{tags[-1]['name']}} </a></span>
```
Comment (gentlecat) [inline critiquebrainz/frontend/templates/macros.html]: I don't understand what you are trying to do here. Why have `for tag in tags[: -1]` loop instead of `for tag in tags`?

File: critiquebrainz/frontend/templates/macros.html [hunk 11]
```diff
@@ -93,8 +93,19 @@
...
+  {% else %}
+  <p>{{ _('(none)') }}</p>
+  {% endif %}
+{% endmacro %}
+
 {% macro soundcloud_player(soundcloud_url) %}
-  <div class=soundcloud-player">
+  <div class="soundcloud-player">
```
Comment (gentlecat) [inline critiquebrainz/frontend/templates/macros.html]: This is unrelated to an actual PR, better put it into a separate commit.

File: critiquebrainz/frontend/views/release_group.py [hunk 12]
```diff
@@ -15,6 +15,9 @@ def entity(id):
     release_group = musicbrainz.get_release_group_by_id(id)
     if not release_group:
         raise NotFound(gettext("Sorry, we couldn't find a release group with that MusicBrainz ID."))
+    tags = None
+    if (release_group.get('tag-list')):
```
Comment (gentlecat) [inline critiquebrainz/frontend/views/release_group.py]: Statement inside `if` doesn't need to be wrapped in parentheses.

File: critiquebrainz/frontend/templates/macros.html (hunk 10 above)
Comment (ferbncode) [inline critiquebrainz/frontend/templates/macros.html]: I was trying to show all tags with a comma except the last tag. Should I try it some other way? (CSS, maybe)

Comment (gentlecat) [inline critiquebrainz/frontend/templates/macros.html]: See `loop.last` in http://jinja.pocoo.org/docs/2.9/templates/#for.

File: critiquebrainz/frontend/templates/macros.html [hunk 13]
```diff
@@ -93,6 +93,19 @@
   </div>
 {%- endmacro %}
 
+{% macro show_tags(tags) %}
+  {% if tags %}
+    {% for tag in tags %}
+      <span><a href="{{"https://musicbrainz.org/tag/%s"|format(tag['name'])}}">{{tag['name']}}</a></span>
```
Comment (gentlecat) [inline critiquebrainz/frontend/templates/macros.html]: What's the `<span>` tag for?

Comment (ferbncode) [inline critiquebrainz/frontend/templates/macros.html]: I was under the conception that ```a``` was not inline. My bad 😅

File: critiquebrainz/frontend/templates/macros.html [hunk 14]
```diff
@@ -93,6 +93,19 @@
...
+{% macro show_tags(tags) %}
+  {% if tags %}
+    {% for tag in tags %}
+      <a href="{{"https://musicbrainz.org/tag/%s"|format(tag['name'])}}">{{tag['name']}}</a>
+      {%- if not loop.last -%}
+        ,
+      {% endif %}
+    {% endfor %}
+  {% else %}
+    <p>{{ _('(none)') }}</p>
```
Comment (gentlecat) [inline critiquebrainz/frontend/templates/macros.html]: You can also add a `class="text-muted"` to this one to make it look a bit better.

**Comment by ferbncode** [comment]
@gentlecat, about the margin and the font-size, I modified ```main.less``` by adding the class ```tags``` which takes care of the ```margin-bottom``` and the ```font-size```. I kept the ```font-size``` as `12px`, similar to tags font in MB. Is it preferred to be done some other way? I would from now take care of the syntax pattern. Thanks 🙂
